src/EditorFeatures/VisualBasicTest/Formatting/Indentation/SmartIndenterTests.vb

```diff
@@ -2841,36 +2841,17 @@ End Class"
             Dim code = "
 Class C
     Sub M()
-        Console.WriteLine(""{0} + {1}"",
-            19, 23 +
-                19,
-
-    End Sub
-End Class"
-
-            AssertSmartIndent(
-                code,
-                indentationLine:=6,
-                expectedIndentation:=12)
-        End Sub
-
-        <WorkItem(3293, "https://github.com/dotnet/roslyn/issues/3293")>
-        <WpfFact, Trait(Traits.Feature, Traits.Features.SmartIndent)>
-        Public Sub TestSmartIndentInArgumentLists4()
-            Dim code = "
-Class C
-    Sub M()
-        Console.WriteLine(""{0} + {1}"",
-            19, 23,
-                19,
+        Method(a +
+          b, c +
+          d,
 
     End Sub
 End Class"
 
             AssertSmartIndent(
                 code,
                 indentationLine:=6,
-                expectedIndentation:=16)
+                expectedIndentation:=13)
```

Comment:
:bulb: This is super weird IMO. It would be better to filter arguments to cases where they start on a line. If not in this pull request, then it would be good to file a bug to clean this up "eventually". 😄

Reply:
If I understand the system correctly, filtering to just arguments that start a line would be even less predictable. This would leave gaps in the argument list where there isn't an indent block operation. In those cases, the indenter would move up the tree to the next indent block operation, which would likely be some node above the argument list. Here's what I *think* would happen:

``` VB
Sub M()
    Method(a +
      b, c +
      d,

End Sub
```

If there isn't an indent block operation created for `c + d`, the indentation would be 4 because the indent block operation from `Sub M() .. End Sub` would be used.

@heejaechang, please correct me if I'm not understanding the code correctly.

Reply:
@sharwell: I think the right experience is to cause the first line break in an argument list after the first argument to use the same indent as the first argument. Then, subsequent line breaks would just use the previous line's indentation. However, I couldn't find a way to achieve that in this system. 😞

Reply:
Skipping arguments because they don't start a line would be equivalent to skipping trailing arguments on the same line. The indentation block from the previous (or first) argument would be used. For the case above, pressing <kbd>Enter</kbd> after `d` would indent to match `a`.

Reply:
Ah yes, you're correct. :smile: I think that'd still be a weird experience though.

``` VB
Sub M()
    Method(a +
      b, c +
      d,
           |
End Sub
```

Reply:
smart indentation, first use formatting rule to find indentation, after that it uses simple indentation logic to find out where to put caret.
(http://index/?leftProject=Microsoft.CodeAnalysis.EditorFeatures&leftSymbol=poqp1cwzffsy&rightProject=Microsoft.CodeAnalysis.CSharp.EditorFeatures&file=Formatting%5CIndentation%5CCSharpIndentationService.Indenter.cs&line=136)

the simple indentation logic does either following previous line or add 4 more spaces.

the special formatting rule for argument, parameter was added for VB since vb had this behavior where caret aligned to first argument/parameter in the list rather than following previous line indentation or 4 spaces.

ex) void method(A b, 
                           | <= here

if one wants this kind of behavior
void method(A b,
    | <= here

one can just remove formatting rule, and let default smart indentation logic to take over.

if one wants to use formatting rule for it, you can use this
http://index/?query=FormattingRu&rightProject=Microsoft.CodeAnalysis.Workspaces&file=Formatting%5CRules%5COperations%5CIndentBlockOption.cs&line=16 and set something like "," as base token, but formatter was not designed to format expressions. it is designed to leave expression exactly as they were as much as possible, so I dont think there is enough options or knobs around expressions.

I think if behavior desired is just letting caret to be either aligned with previous line or 4 more spaces after previous line, it would be better just to let default smart indent behavior to take care of it.

Reply:
Thanks @heejaechang. I think the right behavior for VB probably is:

1. Align with the first argument on the second line of an argument list.
2. Align with the previous line for subsequent lines in an argument list.

If I'm understanding you, removing the current VB formatting rule wouldn't provide that experience, correct?

Reply:
yep. I think your change is general improvement around this area. I was just saying if we want to make it even better, we can tweak it a bit more.